templates/terraform/examples/cloudrun_service_basic.tf.erb

```diff
@@ -0,0 +1,28 @@
+resource "google_cloudrun_service" "<%= ctx[:primary_resource_id] %>" {
+  name          = "<%= ctx[:vars]['cloudrun_service_name'] %>"
+  location = "us-central1"
+  provider = "google-beta"
+
+  metadata {
+    namespace = "<%= ctx[:test_env_vars]['namespace'] %>"
+  }
+
+  spec {
+    containers {
+      image = "gcr.io/cloudrun/hello"
+    }
+  }
+}
+
+# The Service is ready to be used when the "Ready" condition is True
```

Comment:
We can't wait in the resource itself because of the comment that you made about Unknown sometimes being a terminal state, right? Is there any way we can give feedback on that? I feel like doing it this way just pushes the problem down to our users.

Reply:
Summary from in person conversation: It sounds like DomainMapping _should_ be the exception and my understanding so far is that `UNKNOWN` should be a truly transitional state in most cases. We decided that Service needs this before it's really usable so I filed https://github.com/terraform-providers/terraform-provider-google/issues/4091 and I am opening a bug against the DomainMapping endpoint to see if we can figure out a better state resolution.